Добавил FileBackedTaskManager + загрузка/сохранение

mrAlikulov opened the pull request:
Финальное задание спринта 8

mrAlikulov committed:
Добавил FileBackedTaskManager + загрузка/сохранение

EugeneLenkevich commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -0,0 +1,279 @@
+        if (task.getStartTime() != null && task.getEndTime() != null && hasIntersection(task)) {
+
+            tasks.remove(task.getId());
+            throw new IllegalArgumentException("Невозможно создать задачу — пересечение по времени с другой задачей.");
```
IllegalArgumentException - это непроверяемое исключение, ошибка в самой логике. Тут же у нас вполне себе штатная ситуация из нашего бизнес-процесса когда у задач пересеклось время, давай сделаем свое проверяемое (extends Exception) исключение специально для этого случая

mrAlikulov replied:
добавлен TimeOverlapException

EugeneLenkevich replied:
А ты запушил изменения, новых комитов нету?

mrAlikulov replied:
Да запушил

EugeneLenkevich replied:
В ветке вижу только первоначальный комит, точно все изменения залиты, посмотри, пожалуйста

<img width="1224" height="545" alt="image" src="https://github.com/user-attachments/assets/5751128d-13ae-4bb3-89de-af47cff665e6" />

mrAlikulov replied:
все исправил

EugeneLenkevich reviewed (changes requested):
Привет, Джамол! Спасибо за PR.

Во, супер, один только моментик - и исключением, посмотри, пожалуйста :)

mrAlikulov committed:
Добавил FileBackedTaskManager + загрузка/сохранение
Добавлен TimeOverlapException + обновлен InMemoryTaskManager + обновлен FileBackedTaskManager

mrAlikulov commented:
добавлен TimeOverlapException

mrAlikulov committed:
Добавил FileBackedTaskManager + загрузка/сохранение
Добавлен TimeOverlapException + обновлен InMemoryTaskManager + обновлен FileBackedTaskManager
добавлены throws TimeOverlapException в тестах

mrAlikulov committed:
Добавил FileBackedTaskManager + загрузка/сохранение
Добавлен TimeOverlapException + обновлен InMemoryTaskManager + обновлен FileBackedTaskManager
добавлены throws TimeOverlapException в тестах

EugeneLenkevich reviewed (approved):
Во, теперь все на месте, спасибо!
